lib/dataloader/ecto.ex

```diff
@@ -752,13 +752,21 @@ if Code.ensure_loaded?(Ecto) do
            ) do
         [{owner_join_key, owner_key}, {related_join_key, related_key}] = assoc.join_keys
 
-        query
-        |> join(:inner, [x], y in ^assoc.join_through,
-          on: field(x, ^related_key) == field(y, ^related_join_key)
-        )
-        |> where([..., x], field(x, ^owner_join_key) == field(parent_as(:parent), ^owner_key))
+        join =
+          query
+          |> join(:inner, [x], y in ^assoc.join_through,
+            on: field(x, ^related_key) == field(y, ^related_join_key)
+          )
+          |> where([..., x], field(x, ^owner_join_key) == field(parent_as(:parent), ^owner_key))
+
+        binds = Ecto.Query.Builder.count_binds(join)
+
+        join
+        |> Ecto.Association.combine_joins_query(assoc.where, 0)
```

Comment:
Probably worth posting to the ecto mailing list (or just send a PR) to see if we can get `Ecto.Association` to be a documented API.

https://github.com/elixir-ecto/ecto/blob/v3.5.8/lib/ecto/association.ex#L30-L31